### ISSUE TITLE
Support for additional properties disabled in JSON schema

### ISSUE THREAD

**Issue by apmccartney** [spdx/spdx-spec#650]
As currently written, the JSON schema does not allow for additional properties to accommodate extended vendor-specific data. This appears to be at odds with the [SPDX Use Cases 2.0](https://wiki.spdx.org/view/Technical_Team/Use_Cases/2.0), specifically section 16.1, *Communicate data beyond what is described in spec*. 

Earlier versions of the schema did allow for this behavior, but that feature was disabled in commit [a08ffa4](https://github.com/spdx/spdx-spec/commit/a08ffa4b883b35574ae0df8f4566cd031341ba3e).

**Comment by kestewart** [comment]
@goneall - what's been done in 2.3 schema now that it's generated?

**Comment by goneall** [comment]
@kestewart Nothing has changed - as @apmccartney accurately describes, we added restrictions to the JSON schema which limited classes and properties to what was defined in the SPDX spec.

This is consistent with the other schemas.

@apmccartney - the Annotation and comments properties are commonly used to add additional data and support the referenced use case.

That being said, I'm personally open to relaxing the restrictions, but we should discuss this on one of the tech team calls and come to a consensus.

**Comment by kestewart** [comment]
@apmccartney - can you add this to an upcoming tech call agenda at: https://spdx.swinslow.net/p/spdx-tech-minutes and plan to participate in the discussion?

**Comment by goneall** [comment]
Moving to 3.1 for consideration.

cc: @JPEWdev

**Comment by JPEWdev** [comment]
FWIW, my though is that the `Extension` mechanism is what should be used for additional information about Elements, and the JSON schema should retain `"additionalProperties" : false` (or in the newer generated schema, `"unevaluatedProperties": false`). The reason for this is that it signals to users if they mistype a field name, since it will fail schema validation in that case. Otherwise (for non mandatory fields) the error would go unnoticed. Also, it allows us to add more properties to the core objects without needing to worry about conflicting with properties unknown to us added by other parties.

**Comment by goneall** [comment]
> FWIW, my though is that the `Extension` mechanism is what should be used for additional information about Elements, and the JSON schema should retain `"additionalProperties" : false` (or in the newer generated schema, `"unevaluatedProperties": false`). The reason for this is that it signals to users if they mistype a field name, since it will fail schema validation in that case. Otherwise (for non mandatory fields) the error would go unnoticed. Also, it allows us to add more properties to the core objects without needing to worry about conflicting with properties unknown to us added by other parties.

Makes sense.  @apmccartney - pls. take a look at the [Extension class](https://spdx.github.io/spdx-spec/v3.0/model/Extension/Classes/Extension/) in 3.0 and if that satisfies your use case we can close this issue.

**Comment by apmccartney** [comment]
Apologize for my lack communication. I've transitioned employers since initially filing this issue. I'll contact a former colleague who's in a better position to represent my former employer's use case.

**Comment by bact** [comment]
Are we good to close this issue?
